arkanoid/js/ball.js

```diff
@@ -0,0 +1,91 @@
+function Ball(x, y) {
+  this.velocity = 6;
+  this.dx = 0;
+  this.dy = 0;
+  this.x = 440;
+  this.y = 440;
+  this.width = 40;
+  this.height = 40;
+
+  this.start = function () {
+    this.dy = -this.velocity;
+    this.dx = this.getRandomDirection(-this.velocity, this.velocity);
+  };
+
+  this.fly = function () {
+    if (this.dy) {
+      this.y += this.dy;
+    }
+    if (this.dx) {
+      this.x += this.dx;
+    }
+  };
+
+  this.collide = function (element) {
+    var x = this.x + this.dx;
+    var y = this.y + this.dy;
+
+    if (
+      x + this.width > element.x && // LEFT
+      x < element.x + element.width && //RIGHT
+      y + this.height > element.y && // BOTTOM
+      y < element.y + element.height // TOP
+    ) {
+      return true;
+    }
+    return false;
+  };
+
+  this.bumpBlock = function (block) {
+    this.dy *= -1;
+    this.destroyBlock(block);
+  };
+
+  this.destroyBlock = function (block) {
+    block.active = false;
```

Comment:
Manipulating Block's internal properties is sole Block's responsibility and should not be handled by any other class, but the block itself.
To put it simple, once there's collision between the ball and the block, the ball should only react within it's own properties (in this case, only change direction). While block should react on it's own (it should have it's own ".destroy()" method, that should be called next to 'this.ball.bumpBlock' as a separate call).
Otherwise, the unneeded artificial coupling is created between 2 originally independent classes, that results into harder refactoring in future (that will also get worse along with project scaling)